R/plot-build.R

```diff
@@ -189,8 +189,17 @@ ggplot_gtable.ggplot_built <- function(data) {
     position <- "none"
   } else {
     # these are a bad hack, since it modifies the contents of viewpoint directly...
-    legend_width  <- gtable_width(legend_box)
-    legend_height <- gtable_height(legend_box)
+
+    if (any(unitType(legend_box$widths) == "sum")) {
+      legend_width  <- unit(1, "npc")
+    } else {
+      legend_width <- gtable_width(legend_box)
+    }
+    if (any(unitType(legend_box$heights) == "sum")) {
+      legend_height <- unit(1, "npc")
+    } else {
+      legend_height <- gtable_height(legend_box)
+    }
```

Comment:
This is step 3

Reply:
Maybe we could make this a bit more explicit? Can we attach an attribute or something to the unit vector when we do a stretch, and then check for that instead?